File: problems/645-set-mismatch/solution.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int n = nums.length;
        int[] count = new int[n + 1];
        int duplicate = 0;
        int missing = 0;
        long sum = 0;
        long expectedSum = (long) n * (n + 1) / 2;

        for (int num : nums) {
            count[num]++;
            sum += num;
        }

        for (int i = 1; i <= n; i++) {
            if (count[i] == 2) {
                duplicate = i;
            }
            if (count[i] == 0) {
                missing = i;
            }
        }
        
        return new int[]{duplicate, missing};
    }
}

File: problems/645-set-mismatch/solution.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        long long n = nums.size();
        long long sum = 0;
        long long expectedSum = n * (n + 1) / 2;
        unordered_map<long long, long long> count;
        for (long long num : nums) {
            sum += num;
            count[num]++;
        }
        long long duplicate = 0;
        for (auto const& [key, val] : count) {
            if (val > 1) {
                duplicate = key;
                break;
            }
        }
        long long missing = expectedSum - (sum - duplicate);
        return {duplicate, missing};
    }
};